src/ImageEditor.zig

```diff
@@ -18,3 +19,71 @@ pub fn flipVertically(pixels: *const color.PixelStorage, height: usize, allocato
         @memcpy(row2_data, temp);
     }
 }
+
+/// Create and allocate a cropped subsection of this image.
+pub fn crop(
+    image: *const ImageUnmanaged,
+    crop_area: Box,
+    allocator: std.mem.Allocator,
+) Error!ImageUnmanaged {
+    const box = crop_area.normalize(image.width, image.height);
+
+    const new_buffer = try color.PixelStorage.init(
```

Comment:
Indexed pixels are not supported by this function. You'll just need to copy the palette to the new buffer, you can use `PixelFormat.isIndexed()` to known when the source pixel format is indexed.

Also not sure if `grayscale8Alpha` and `grayscale16Alpha` would pass as well. Another reason why we needs tests for the crop.